### DISCUSSION TITLE
pip install mpi4py can't find openmpi or mpich files.

### DISCUSSION THREAD

**Discussion by LeamHall** [mpi4py/mpi4py#579] [Q&A] [answered]
Just learning about MPI and mpi4py so it may be operator error. 

I am trying to install mpi4py in a Python 3.12 venv on a Fedora/RH host and the first failure was that it couldn't find mpi.h. 
1. I installed openmpi and openmpi-devel  (4.1.4) on the host, still failed.
2. I installed mpich and mpich-devel (4.0.2), still failed.
3. I soft linked /usr/include/openmpi-x86_64 to ~/venv/include/openmpi-x86_64, still failed.
4. I copied /usr/include/openmpi-x86_64/mpi.h to ~/venv/include, and it failed while looking for mpi_portable_platform.h.
5. I copied mpi_portable_platform.h to ~/venv/include, and now _configtest.c fails with undefined references to MPI_Init and MPI_Finalize

Obviously I'm doing something wrong, just not sure what. Suggestions?

**Comment by dalcinl** [comment, marked as answer]
You may need to read how MPI is managed in Fedora. https://docs.fedoraproject.org/en-US/neurofedora/mpi/

After installing MPICH and/or Open MPI, please close your terminal and open another one. In the new terminal, load the module of the MPI you want to use and check the `mpicc` and `mpiexec` commands are found:
```console
$ module load mpi/mpich-x86_64 
$ which mpicc
/usr/lib64/mpich/bin/mpicc
$ which mpiexec
/usr/lib64/mpich/bin/mpiexec
```
Now you should be able to `pip install mpi4py`.

**Reply by LeamHall** [reply]
@dalcinl that worked, thanks!
